### ISSUE TITLE
Unsubscribe doesn't close the connection in every case

### ISSUE THREAD

**Issue by akarelas** [insidewhy/rxjs-websockets#126]
Here's an example.

On a Linux terminal, type this to open a TCP listener on port 5362 that accepts connections and doesn't send anything nothing:
```bash
nc -l 5362
```

Then execute this code from a browser:
```javascript
// example 1: browser "Firefox can’t establish a connection" error doesn't
// appear in 2 seconds, but in 15 (wrong behaviour - what if it managed to
// connect somewhere in between? I would end up with an unused websocket
// connection forever, which would burden the server)
console.log('starting example 1...')
let s = makeWebSocketObservable('ws://localhost:5362').subscribe()
setTimeout(() => s.unsubscribe(), 2e3)

// example 2: browser error "The connection to ws://localhost:5362/ was interrupted"
// appears in 2 seconds (which I think is the correct behaviour)
setTimeout(() => {
    console.log('starting example 2...')
    var ws = new WebSocket('ws://localhost:5362')
    setTimeout(() => ws.close(), 2e3)
}, 30e3)
```

**Comment by insidewhy** [comment]
@akarelas I'm guessing this happens because the `socket.onopen` method is never called, which in turn means `isSocketOpen = true` never gets run. Maybe that logic just needs to be deleted.

**Comment by insidewhy** [comment]
i.e. I guess it's fine (and desirable) to call `socket.close()` even before the `open` event is fired.

**Comment by insidewhy** [comment]
@akarelas Thanks for the report, released `8.0.1` with the fix.
